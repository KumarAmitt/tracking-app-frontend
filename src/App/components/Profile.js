import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import ExitToAppOutlinedIcon from '@material-ui/icons/ExitToAppOutlined';
import { getSessionInfo, loadSession } from '../../store/slicers/user_session';
import { logoutUser } from '../../store/slicers/user_logout';
import AppBar from './AppBar';
import './style/Profile.css';

const Profile = () => {
  const dispatch = useDispatch();
  const sessionInfo = useSelector(getSessionInfo);

  useEffect(() => {
    dispatch(loadSession());
  }, []);

  const handleLogout = async () => {
    await dispatch(logoutUser());
    dispatch(loadSession());
  };

  // console.log(sessionInfo);
  // if (!sessionInfo.logged_in) {
  //   return <Redirect to="/" />;
  // }
  if (!sessionInfo.logged_in) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <AppBar title="Profile" />
      <div className="profile-top">
        <h2>{sessionInfo.user.username}</h2>
      </div>
      <button type="button" onClick={handleLogout} className="logout-button">
        <ExitToAppOutlinedIcon />
        <div>Sign Out</div>
      </button>
    </>
  );
};

export default Profile;
