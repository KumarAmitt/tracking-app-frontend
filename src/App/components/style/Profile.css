.profile-top {
  font-size: 2rem;
  font-weight: bold;
  background-color: var(--accent-primary);
  height: 40vh;
  display: flex;
  justify-content: flex-end;
  flex-direction: column;
}

.profile-top h2 {
  text-align: center;
  color: #eee;
}

.logout-button {
  margin-top: 2em;
  margin-left: 1em;
  display: flex;
  align-items: center;
  padding: 0.5em 1em;
  border: none;
  background-color: transparent;
  font-weight: bold;
  color: red;
  transition: all linear 300ms;
  cursor: pointer;
}
.logout-button div {
  margin-left: 0.5em;
}

.logout-button:hover {
  transform: scale(1.1);
}