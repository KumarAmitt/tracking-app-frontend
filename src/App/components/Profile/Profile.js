import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import ExitToAppOutlinedIcon from '@material-ui/icons/ExitToAppOutlined';
import { getSessionInfo, loadSession } from '../../../store/slicers/userSession';
import { logoutUser } from '../../../store/slicers/userLogout';
import AppBar from '../AppBar/AppBar';
import './style/Profile.css';

const Profile = () => {
  const dispatch = useDispatch();
  const sessionInfo = useSelector(getSessionInfo);

  useEffect(() => {
    dispatch(loadSession());
  }, []);

  const handleLogout = async () => {
    await dispatch(logoutUser());
    dispatch(loadSession());
  };

  if (!sessionInfo.logged_in) {
    return <Redirect to="/" />;
  }

  return (
    <>
      <AppBar title="Profile" />
      <div className="profile-top">
        <h2>{sessionInfo.user.username}</h2>
      </div>
      <button type="button" onClick={handleLogout} className="logout-button">
        <ExitToAppOutlinedIcon />
        <div>Sign Out</div>
      </button>
    </>
  );
};

export default Profile;
