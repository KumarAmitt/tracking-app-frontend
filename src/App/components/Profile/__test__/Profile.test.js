import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../store/configureStore';
import Profile from '../Profile';

const store = configureAppStore();

describe('Profile', () => {
  it('should match the snapshot of Profile component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
