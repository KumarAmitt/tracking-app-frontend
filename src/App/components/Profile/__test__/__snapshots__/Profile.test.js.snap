// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match the snapshot of Profile component 1`] = `null`;
